observability/tracing.go

```diff
@@ -171,29 +173,76 @@ func getExporter(collectorURL string) (sdktrace.SpanExporter, error) {
 					)
 				} else {
 					log.Debugf("Creating OpenTelemetry collector with URL https://%s", collectorURL)
-
-					client = otlptracehttp.NewClient(otlptracehttp.WithEndpoint(collectorURL),
-						tracingOptions,
-					)
+					// That's mainly for testing
+					if tracingOpt.Otel.SkipVerify {
+						log.Trace("OpenTelemetry collector will not verify the remote certificate")
+						client = otlptracehttp.NewClient(otlptracehttp.WithEndpoint(collectorURL),
+							otlptracehttp.WithTLSClientConfig(&tls.Config{InsecureSkipVerify: true}),
+							tracingOptions,
+						)
+					} else {
+						client = otlptracehttp.NewClient(otlptracehttp.WithEndpoint(collectorURL),
+							tracingOptions,
+						)
+					}
 				}
 
 				ctx := context.Background()
-				exporter, err := otlptrace.New(ctx, client)
-				return exporter, err
+				httpExporter, err2 := otlptrace.New(ctx, client)
+				return httpExporter, err2
```

Comment:
minor nit that you don't have to fix: large nested if/else blocks are hard to read and follow. e.g.
```
else {
			return nil, fmt.Errorf("Error in configuration options getting the observability exporter. Invalid collector type [%v].", tracingOpt.CollectorType)
		}
```
can be moved to a check at the beginning:
```
if tracingOpt.CollectorType != OTEL || tracingOpt.CollectorType != JAEGER {
  return nil, fmt.Errorf(...)
}
```
Most of the nesting in this function can probably be collapsed that way. Validating the config opts might also belong in the `config` package.

Reply:
Updated.